docs/USAGE-GUIDE.md

```diff
@@ -1365,7 +1365,20 @@ ssize_t s2n_client_hello_get_extension_by_id(struct s2n_client_hello *ch, s2n_tl
 - **out** Pointer to a buffer into which the extension bytes should be copied.
 - **max_length** Max number of bytes to copy into the **out** buffer.
 
-**s2n_client_hello_get_extension_length** returns the number of bytes the given extension type takes on the ClientHello message received by the server; it can be used to allocate the **out** buffer.
+**s2n_client_hello_get_extension_length** will return `0` if the extension is present but empty and also if the extension is not present. s2n_errno can be used to decipher between these two cases:
```

Comment:
Is this guaranteed to work? I think we make the same promises about s2n_errno that posix methods make about errno: it'll only be meaningful if the method indicates failure. Otherwise, it could be anything.

Reply:
I'm updating the docs saying s2n is committing for this to work. If we aren't going to commit to this working, then this api is not usable.

Reply:
Can we enforce it in code then? I know posix methods don't really set errno to 0, but we already do that in a couple places in S2N. Can we do that [here](https://github.com/aws/s2n-tls/blob/main/tls/s2n_client_hello.c#L547) to make sure s2n_errno really is 0 on success?  Otherwise we have to make sure s2n_client_hello_get_parsed_extension and any of its dependencies don't set s2n_errno on success.

Also probably worth noting the required behavior in a comment on the method itself, in case someone editing it doesn't read the documentation.

Reply:
> we make the same promises about s2n_errno that posix methods make about errno

is there a way we can communicate the present but empty without relying on errno? maybe a separate api? having this as an exception is going to make things more confusing in the long run for developers and also users.